components/camel-dynamic-router/src/main/java/org/apache/camel/component/dynamicrouter/DynamicRouterEndpoint.java

```diff
@@ -140,30 +149,87 @@ protected void doInit() throws Exception {
         super.doInit();
```

Comment:
Is all this code in the constructor? If so it should be moved to doInit where you do all kind of initialization

Reply:
Apologies, but I am not sure what you mean.  It looks like you are referring to the code in `doInit()`, so I must be misunderstanding you.  Could you please clarify?